packages/react-scripts/template/README.md

```diff
@@ -2196,6 +2196,67 @@ If you are not using the HTML5 `pushState` history API or not using client-side
 
 This will make sure that all the asset paths are relative to `index.html`. You will then be able to move your app from `http://mywebsite.com` to `http://mywebsite.com/relativepath` or even `http://mywebsite.com/relative/path` without having to rebuild it.
 
+#### Building for Multiple Environments
+
+Applications are generally split between different environments such as staging, production, and development. To allow the app to run in these different environments one must set environment variables to be able to conditionally run different processes depending on the specified environment.
+
+`create-react-app` handles environment variables in a specific way. [Adding Custom Environment Variables](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#adding-custom-environment-variables)
+
+1. You cannot override `NODE_ENV`, it is always set to `'production'`
+
+2. It is mandated that you prefix any custom environment variables with `REACT_APP_`
+
+    - This means you cannot run:
+
+      - `NODE_ENV=staging npm run build`
+
+    - But you can run:
+
+      - `REACT_APP_NODE_ENV=staging npm run build`
```

Comment:
I think using `NODE_ENV` and `REACT_APP_NODE_ENV` for this example is potentially confusing. The similar names implies that there is some connection between them. The example we commonly use is something like `REACT_APP_API_URL` which is set to one value in `.env.production` or `.env` and then overridden for other environments.